### DISCUSSION TITLE
blog/retrospect/2-abstract-architecture

### DISCUSSION THREAD

**Discussion by giscus** [pySoo/Next-blog#73] [General]
# blog/retrospect/2-abstract-architecture

실무에서는 요구사항이 변화되는 상황을 자주 마주하게 됩니다. 변화에 유연한 설계를 하려면 어떤 점들을 고려하는게 좋을까요?

https://enjoydev.life/blog/retrospect/2-abstract-architecture

**Comment by Songhee99** [comment]
추상화에 대한 개념이 모호하게 잡혀있었는데, 이 글 덕분에 좀 더 정리되었습니다. 감사합니다~

**Reply by pySoo** [reply]
긴 글 읽어주셔서 감사해요 ㅎㅎ 도움이 되었다면 다행입니다! ☺️

**Comment by loevray** [comment]
항상 고민해도 정답이 없는 주제라 머리가 아팠는데, 친절하게 설명해주신 글 덕분에 도움이 됐습니다
그리고 기존 엘리먼트 확장할때 `HTMLAttributes`를 많이 사용했었는데, `ComponentProps`란 것도 있었다니...😮
많이 배워갑니다 ㅎㅎ

**Reply by pySoo** [reply]
오 영현님 안녕하세요!! 긴 글 읽어주셔서 감사드립니다 😊
맞아요 정답이 없기 때문에 상황마다 적절한 방법을 떠올리는 것이 어려운 것 같아요 😵‍💫 그래도 도움이 되었다니 다행입니다..!

조금 덧붙이자면 HTMLAttributes 내부에서는 제네릭을 사용하지 않아서 textarea의 maxLength와 같이 특정 엘리먼트에 있는 속성은 사용할 수 없다고 해요..!
저도 최근에 알게 되어서 컴포넌트 확장시에는 의식적으로 ComponentProps를 사용해보려고 합니다 ㅎㅎ 👍
